src/notebooklm/_retry.py

```
@@ -0,0 +1,333 @@
+                            "Rate limited — using server-suggested retry-after "
+                            "of %.1fs (attempt %d/%d)",
+                            delay,
+                            attempt,
+                            max_attempts,
+                        )
+                    elif attempt < max_attempts:
+                        # Apply exponential backoff with optional jitter
+                        if jitter:
+                            delay = min(delay * backoff_factor, max_delay)
+                            # Jitter: random fraction in [0.75, 1.25] of the delay
+                            delay *= 0.75 + random.random() * 0.5
```

Ruff (0.15.9)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)